### PR TITLE
When toggling digitizing on, insure an editable layer is selected

### PR DESCRIPTION
**Pull request by nirvn**
This fixes a UX issue I've heard by many people trying QField (from beginners all the way to pros). It goes like this: "Hey, I opened the demo bees project, toggled edit mode on, and I couldn't see any changes in the interface".

The PR adds a function to auto-select the first _editable_ layer when toggling digitizing on if no layer is selected or if the selected layer is locked (read-only or geometry locked). It's a _huge_ UX improvement. 

GIF:
Peek 2020-12-25 20-12

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for bd97ff7d45743e007fdbe1d21ff80dfa6d5d7afb:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1529-c05db98d2f787a6027a3f7206af65ffb732d3cb4-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1529-c05db98d2f787a6027a3f7206af65ffb732d3cb4-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1529-c05db98d2f787a6027a3f7206af65ffb732d3cb4-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1529-c05db98d2f787a6027a3f7206af65ffb732d3cb4-x86.apk)

File: src/qml/qgismobileapp.qml [hunk 1]
```diff
@@ -115,7 +115,12 @@ ApplicationWindow {
         displayToast( qsTr( 'You are now in browse mode' ) );
         break;
       case 'digitize':
-        displayToast( qsTr( 'You are now in digitize mode' ) );
+        dashBoard.ensureEditableLayerSelected();
+        if (dashBoard.currentLayer) {
+          displayToast( qsTr( 'You are now in digitize mode on layer %1' ).arg( dashBoard.currentLayer.name ) );
+        } else {
```
Comment (m-kuhn) [inline src/qml/qgismobileapp.qml:121]:
```suggestion
        }
        else
        {
```

File: src/qml/qgismobileapp.qml [hunk 2]
```diff
@@ -115,7 +115,12 @@ ApplicationWindow {
         displayToast( qsTr( 'You are now in browse mode' ) );
         break;
       case 'digitize':
-        displayToast( qsTr( 'You are now in digitize mode' ) );
+        dashBoard.ensureEditableLayerSelected();
+        if (dashBoard.currentLayer) {
```
Comment (m-kuhn) [inline src/qml/qgismobileapp.qml:119]:
```suggestion
        if (dashBoard.currentLayer)
        {
```

**Comment by m-kuhn** [comment]
Nice!!

I need to try qmlformat with qt 6 again.

**Review by m-kuhn — APPROVED** [review]

**Comment by nirvn** [comment]
@m-kuhn , do we have some rules on formatting for qml js? Should it match our c++ style as much as possible? I was deliberately trying to break away from our if (...)<newline>{ habit here :)

**Comment by m-kuhn** [comment]
Ideally we follow what qmlformat does, for the sake of consistency.
Unfortunately when I looked last comments where handled quiet badly

**Comment by VxTedxV** [comment]
Not a good idea for beginners : they could write in the wrong layer (choosen by default without any choice from the user)... But I'll admit it's a good idea for advanced users.

**Comment by nirvn** [comment]
@VxTedxV , most of the interactions I've had with new QField users disagrees with your statement :) The majority of new users are confused by the absence of any UI change when toggling the digitizing on. Also, say someone would have selected a layer while randomly testing the interface and toggle the digitizing on, he/she would still risk entering data in a "wrong" layer.

Also, on master, we're doing a much better job at giving queues as to which layer is going to be edited:
- With this PR, the toaster message that appears when you toggle digitize on actually tells you which layer is being edited (see GIF)
- Prior to this PR, @suricactus made a change so the active / selected layer visual queue is much more visible (green background vs what was before a slightly too subtle green text).

So weighting risks and rewards, I'm very confident this is the right move for QField.

That said, where I think you have a point is how we could do better in introducing QField to newcomers via more hints and explanations when they  trigger QField features for the first time.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for d7c2a8c03b1ba333ee705fca564aa5d8fd478580:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1529-64a38648cd269539b668769f1f11f564ccf401cb-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1529-64a38648cd269539b668769f1f11f564ccf401cb-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1529-64a38648cd269539b668769f1f11f564ccf401cb-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1529-64a38648cd269539b668769f1f11f564ccf401cb-x86.apk)

**Pull request merged**